mockgen/parse.go

```diff
@@ -547,19 +547,44 @@ func packageNameOfDir(srcDir string) (string, error) {
 }
 
 // parseImportPackage get package import path via source file
+// an alternative implementation is to use:
+// cfg := &packages.Config{Mode: packages.NeedName, Tests: true, Dir: srcDir}
+// pkgs, err := packages.Load(cfg, "file="+source)
+// However, it will call "go list" and slow down the performance
 func parsePackageImport(source, srcDir string) (string, error) {
-	cfg := &packages.Config{Mode: packages.LoadFiles, Tests: true, Dir: srcDir}
-	pkgs, err := packages.Load(cfg, "file="+source)
-	if err != nil {
-		return "", err
+	moduleMode := os.Getenv("GO111MODULE")
+	// trying to find the module
+	if moduleMode != "off" {
```

Comment:
Is `auto` a special case?

Reply:
Might be worth being explicit /w a switch

Reply:
the meaning of `auto` changes from Go release to release. To be safe, I only skip checking `go.mod` when the mode is "off". This may result in different behavior than "go list" in older Go releases.

For example, when "GO111MODULE" is "auto", if a directory is under GOPATH, but also has a go.mod file, "go list" will use GOPATH in 1.12, and use Go module in 1.13. This PR is similar to the behavior of 1.13.